### PR TITLE
Adding test script for configmap yaml

### PR DESCRIPTION
**Pull request by shruacha1234**
Signed-off-by: Shruthi <Shruthi.Acharya@ibm.com>

### CODE REVIEW

**Comment by chandrams** [comment]
@shruacha1234 - Can you please resolve the conflicts

File: tests/scripts/configmap_constants.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#
+##### script containing constants for configmap yaml tests #####
+#
+
+# get the absolute path of current directory
+CURRENT_DIR="$(dirname "$(realpath "$0")")"
+pushd ${CURRENT_DIR}/../.. >> /dev/null
+
+# Path to the directory containing yaml files
+configmap="${PWD}/manifests/configmaps"
```
Comment (chandrams) [inline tests/scripts/configmap_constants.sh:25]: I don't think we are having any static yamls, do we need this?

Comment (shruacha1234) [reply tests/scripts/configmap_constants.sh:25]: Its the path to configmap present in autotune manifest ., To copy the same and change it according to the testcase

File: tests/scripts/common_functions.sh [hunk 2]
```diff
@@ -135,6 +139,8 @@ function deploy_autotune() {
 	echo "Deploying autotune"
 	if [ -z "${AUTOTUNE_IMAGE}" ]; then
 		cmd="./deploy.sh -c ${cluster_type}"
+	elif [[ ! -z "${AUTOTUNE_IMAGE}" && ! -z "${CONFIG_DIR}" ]]; then
+		cmd="./deploy.sh -c ${cluster_type} -i ${AUTOTUNE_IMAGE} -d ${CONFIG_DIR}"
 	else
 		cmd="./deploy.sh -c ${cluster_type} -i ${AUTOTUNE_IMAGE}"
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:145]: In the else we will need to pass the config map by changing just the logging level to debug.

File: tests/scripts/configmap_constants.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+"minikube-debug-config")
+
+# String to be searched in minikube configmap
+declare -A configmap_minikube_find
+configmap_minikube_find=([minikube-invalid-cluster-type]='kubernetes'
+[minikube-invalid-k8s-type]='minikube'
+[minikube-invalid-monitoring-agent]='prometheus'
+[minikube-invalid-monitoring-service]='prometheus-k8s'
+[minikube-debug-config]='info')
+# String to be replaced in minikube configmap
```
Comment (chandrams) [inline tests/scripts/configmap_constants.sh:45]: Insert a new line here

File: tests/scripts/configmap_constants.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+[minikube-debug-config]='debug')
+
+# Expected autotune object for minikube configmap
+declare -A configmap_minikube_autotune_objects
+configmap_minikube_autotune_objects=([minikube-invalid-cluster-type]='true' 
+[minikube-invalid-k8s-type]='true'  
+[minikube-invalid-monitoring-agent]='true'
+[minikube-invalid-monitoring-service]='true'
+[minikube-debug-config]='true' )
+# Expected log message for minikube configmap
```
Comment (chandrams) [inline tests/scripts/configmap_constants.sh:60]: Insert a new line here too

File: tests/scripts/configmap_yaml_tests.sh [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+. ${SCRIPTS_DIR}/configmap_constants.sh
+
+# Update the config map yaml with specified field
+function update_yaml() {
+	find=$1
+	replace=$2
+	sed -i 's/'${find}'/'${replace}'/g' ${config_yaml}
+}
+
+# application autotune yaml tests
```
Comment (chandrams) [inline tests/scripts/configmap_yaml_tests.sh:33]: Modify the comments to indicate configmap tests

File: tests/scripts/common_functions.sh [hunk 6]
```diff
@@ -133,10 +148,12 @@ function deploy_autotune() {
 	fi
 	
 	echo "Deploying autotune"
-	if [ -z "${AUTOTUNE_IMAGE}" ]; then
-		cmd="./deploy.sh -c ${cluster_type}"
-	else
-		cmd="./deploy.sh -c ${cluster_type} -i ${AUTOTUNE_IMAGE}"
+	if [ -z "${AUTOTUNE_IMAGE}" ! -z "${CONFIGMAP_DIR}" ]; then
+		cmd="./deploy.sh -c ${cluster_type} -d ${CONFIGMAP}"
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:152]: This is still not clear to me or right, can you add some comments here.

File: tests/scripts/common_functions.sh [hunk 7]
```diff
@@ -133,10 +148,12 @@ function deploy_autotune() {
 	fi
 	
 	echo "Deploying autotune"
-	if [ -z "${AUTOTUNE_IMAGE}" ]; then
-		cmd="./deploy.sh -c ${cluster_type}"
-	else
-		cmd="./deploy.sh -c ${cluster_type} -i ${AUTOTUNE_IMAGE}"
+	if [ -z "${AUTOTUNE_IMAGE}" ! -z "${CONFIGMAP_DIR}" ]; then
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:151]: AND condition is missing here

**Review by chandrams — APPROVED** [review]
lgtm

**Pull request merged**